mysql-test/main/path_resolution.result

```diff
@@ -0,0 +1,3006 @@
+#
+USE test;
+CREATE PACKAGE pack
+FUNCTION func() RETURNS TEXT;
+END;
+$$
+CREATE PACKAGE BODY pack
+FUNCTION func() RETURNS TEXT RETURN 'Invoke from Db: test, Pack: pack, Rt: func';
+END;
+$$
+USE pack;
+CREATE FUNCTION func() RETURNS TEXT RETURN 'Invoke from Db: pack, Rt: func';
+USE test;
+SELECT pack.func();
+pack.func()
+Invoke from Db: test, Pack: pack, Rt: func
+DROP PACKAGE BODY pack;
+DROP PACKAGE pack;
+SELECT pack.func();
+pack.func()
+Invoke from Db: pack, Rt: func
+DROP FUNCTION pack.func;
+SELECT pack.func();
+ERROR 42000: FUNCTION pack.func does not exist
+SET @@path = DEFAULT;
+#
+# type 2.13
+# sql_mode: DEFAULT | function | is qualified
+# curr db: test2 | path set: yes
+# resolution priority in between qualified package function vs qualified db function resolution
+#
+USE test;
+CREATE PACKAGE pack
+FUNCTION func() RETURNS TEXT;
+END;
+$$
+CREATE PACKAGE BODY pack
+FUNCTION func() RETURNS TEXT RETURN 'Invoke from Db: test, Pack: pack, Rt: func';
+END;
+$$
+USE pack;
+CREATE FUNCTION func() RETURNS TEXT RETURN 'Invoke from Db: pack, Rt: func';
+USE test2;
+SELECT pack.func();
+pack.func()
+Invoke from Db: pack, Rt: func
+SET PATH= 'sys,test';
+SELECT pack.func();
+pack.func()
+Invoke from Db: test, Pack: pack, Rt: func
+DROP PACKAGE BODY test.pack;
+DROP PACKAGE test.pack;
+SELECT pack.func();
+pack.func()
+Invoke from Db: pack, Rt: func
+DROP FUNCTION pack.func;
+SELECT pack.func();
+ERROR 42000: FUNCTION pack.func does not exist
+DROP DATABASE pack;
+SET @@path = DEFAULT;
+#
+# type 2.14
+# sql_mode: DEFAULT | function | is qualified
+# curr db: X | path set: yes
+# qualified package function invoke internal private function
+#
+SET PATH= 'test,test2';
+USE test2;
+CREATE FUNCTION func_prv() RETURNS TEXT RETURN 'Invoke from Db: test2, Rt: func_prv';
+USE test;
+CREATE FUNCTION func_prv() RETURNS TEXT RETURN 'Invoke from Db: test, Rt: func_prv';
+CREATE PACKAGE pack
+FUNCTION func_pub() RETURNS TEXT;
+END;
+$$
+CREATE PACKAGE BODY pack
+FUNCTION func_prv() RETURNS TEXT RETURN 'Invoke from Db: test, Pack: pack, Rt: func_prv';
+FUNCTION func_pub() RETURNS TEXT RETURN func_prv();
+END;
+$$
+CREATE DATABASE deselect_db;
+USE deselect_db;
+DROP DATABASE deselect_db;
+SELECT pack.func_pub();
+pack.func_pub()
+Invoke from Db: test, Pack: pack, Rt: func_prv
+SET @@path = DEFAULT;
+#
+# type 2.15
+# sql_mode: DEFAULT | function | is qualified
+# curr db: test2 | path set: yes
+# qualified package function invoke internal private function
+#
+SET PATH= 'test,test2';
+USE test2;
+SELECT pack.func_pub();
+pack.func_pub()
+Invoke from Db: test, Pack: pack, Rt: func_prv
+DROP FUNCTION func_prv;
+USE test;
+DROP FUNCTION func_prv;
+DROP PACKAGE BODY pack;
+DROP PACKAGE pack;
+SET @@path = DEFAULT;
+#
+# type 3.1
+# sql_mode: DEFAULT | procedure | is unqualified
+# curr db: X | path set: no
+#
+CREATE PROCEDURE proc() SELECT 'Invoke from Db: test, Rt: proc';
```

Comment:
may be write `'Invoke from Db: test, Rt: proc' as 'log:'` ?
very minor detail, really, don't change if you don't want to